ngraph/test/type_prop/non_zero.cpp

```diff
@@ -47,6 +47,16 @@ TEST(type_prop, non_zero_string_output_type)
         non_zero->get_output_partial_shape(0).same_scheme(PartialShape{4, Dimension::dynamic()}));
 }
 
+TEST(type_prop, non_zero_bool_input_type)
+{
+    auto data = make_shared<op::Parameter>(element::boolean, Shape{1, 2, 3, 4});
+    auto non_zero = make_shared<op::v3::NonZero>(data, element::i32);
+
+    ASSERT_EQ(non_zero->get_output_element_type(0), element::i32);
+    EXPECT_TRUE(
+        non_zero->get_output_partial_shape(0).same_scheme(PartialShape{4, Dimension::dynamic()}));
```

Comment:
Should it be like this?
```suggestion
    EXPECT_EQ(
        non_zero->get_output_partial_shape(0), PartialShape{4, {0, 24}}));
```